### PR TITLE
Feat/16 authentication

### LINKED ISSUE #16: Hook up authentification backend to login form
Implement logic for logging in and out by querying the backend

### PR DESCRIPTION
**Pull request by CamiloGarciaLaRotta**
### Description
 - Advances #10 by implementing error handling of the login form
 - Login is now a reducer component. 
   It performs the authentication and via a callback returns the JWT to the App component.
 - Services has a side-effect method "authenticate" which POSTs the credentials to the Auth backend.
   If the login is succesful, it returns the JWT to Login.
   If the login fails, it changes the state of Login to DisplayError
 - Services has a new type "authResponse" to decode the HTTP response
 - valueFromEvent was refactored to Services because its used in multiple components
 - Changed the `Sign in with Google` dummy button for `Sign Up`,
   which redirects the user to the web-app sign up page 
   (when it will be available online)

Closes #16

### Checklist
- [x] Code compiles correctly
- [x] All tests passing
- [x] There is an open issue linked to this PR
- [ ] Created tests which fail without the change (if possible)
- [ ] Extended the README / documentation (if necessary)

### CODE REVIEW

**Comment by CamiloGarciaLaRotta** [comment]
Travis failed because `Coveralls` hasn't created a record for the branch.
Probably because we have the free version we get super shitty service.
It took the last branch 1h to create the record.

File: manifest.json
Comment (alexboii) [inline manifest.json:14]: why was storage removed? are you not storing the jwt in localstorage?

File: src/ScrapingInput.re
Comment (alexboii) [inline src/ScrapingInput.re:3]: why was this moved from here to services? if anything, it needs to be under a new "utilities.re" file

File: src/App.re
Comment (alexboii) [inline src/App.re]: Option(string) should be used here, but at this point I don't care

**Review by alexboii — CHANGES_REQUESTED** [review]
looks okay, as long as it works

the only thing that's really needed here is that you ought to store the jwt in local storage and load it before the HOC mounts, check what the guy does here: https://github.com/jchavarri/chrome-extension-starter-reason-react/blob/master/src/colorSelector.re

there are bindings for it too

File: manifest.json
Comment (CamiloGarciaLaRotta) [reply manifest.json:14]: thats beyond the scope of hooking the extension to the auth API (#16)
The storage of the JWT is in the scope of #19 
I can make a PR after for with that enhancement

File: src/ScrapingInput.re
Comment (CamiloGarciaLaRotta) [reply src/ScrapingInput.re:3]: I can move it there, in the PR description I said it was extracted because its used across multiple components

File: src/App.re
Comment (CamiloGarciaLaRotta) [reply src/App.re]: addressed in latest commit

File: src/ScrapingInput.re
Comment (CamiloGarciaLaRotta) [reply src/ScrapingInput.re:3]: addressed in latest commit

**Comment by CamiloGarciaLaRotta** [comment]
for reference, many people are experiencing the same Coveralls failure: [Coveralls GitHub Issue](https://github.com/lemurheavy/coveralls-public/issues/632)

File: manifest.json
Comment (alexboii) [reply manifest.json:14]: yeah but wouldn't I have to login every time I click on the chrome extension? the two tasks are hand-in-hand

**Review by alexboii — CHANGES_REQUESTED** [review]
I still think that the jwt storage & login should be the same PR 

for then testing the next part (adding job applications) would become a nightmare 

if you want I can approve this but I say we wait for #19 before merging

File: manifest.json
Comment (CamiloGarciaLaRotta) [reply manifest.json:14]: yes, you currently will have to login every time you click on the extension.
this PR only addresses login in and out for the `Login` components perspective.
#19 handles login in and out from a broader perspective.

this PR ensures that first step is functional in order to work on the next PRs (#17 #19)

Comment (CamiloGarciaLaRotta) [reply manifest.json:14]: I vote against holding up the PR until #19 is done.
storing the JWT in storage has nothing to do with the `Login` component and its call to the Auth API

**Review by alexboii — APPROVED** [review]

**Pull request merged**
